aesara/tensor/elemwise.py

```diff
@@ -1436,9 +1426,8 @@ def _c_all(self, node, name, inames, onames, sub):
         else:
             adtype = odtype
 
-        axis = self.axis
-        if axis is None:
-            axis = list(range(len(input.type.broadcastable)))
+        assert isinstance(axis, TensorConstant)
+        axis = axis.data
```

Comment:
I am currently working on that implementation. Where `axis` is a `varargs` one problem I am running into is how to differentiate between `axis = None` and `axis = ()` The former reduces along all axes but later does nothing. For my current implementation I'm assuming `axis=()` passing nothing in `varargs` in `make_node` means reduction along all axes. 

Note that this is being asked for `Op.make_node` inputs not for the actual apply node inputs which already convert None to corresponding axis representation.

Reply:
If it's along all axes, then there should be an axia value for each axis, like it was doing.  If there are no axes and no reduction can be done, that seems more like invalid input.

Reply:
With #711 , I don't think there will be a need to this anymore. So I guess we can go ahead with this particular implementation, right ?

Reply:
Well, it definitely wouldn't be bad to have the `Op` working with varargs; it's always a safe approach.  Plus, we might be able to merge this before #711, since that involves a few big changes that need more thought and testing.